draft-ietf-jsonpath-base.md

```diff
@@ -1043,29 +1043,32 @@ logical-or-expr  = logical-and-expr *("||" logical-and-expr)
 logical-and-expr = basic-expr *("&&" basic-expr)      ; conjunction
                                                       ; binds more tightly than disjunction
 
-basic-expr   = comp-expr / existence-expr / paren-expr / (neg-op paren-expr)
-existence-expr = [neg-op] path                        ; path existence or non-existence
+basic-expr   = exist-expr / paren-expr / (neg-op paren-expr) / relation-expr
+exist-expr   = [neg-op] path                          ; path existence or non-existence
+path         = rel-path / json-path
+rel-path     = "@" *(dot-selector / index-selector)
 paren-expr   = "(" boolean-expr ")"                   ; parenthesized expression
 neg-op       = "!"                                    ; not operator
 
-comp-expr    = path (comp-op comparable /             ; comparison
-                     regex-op regex     /             ; RegEx test
-                     in-op container)                 ; containment test
-path         = rel-path / json-path
+relation-expr = comp-expr /                           ; comparison test
+                regex-expr /                          ; regular expression test
+                contain-expr                          ; containment test
+
+comp-expr    = comparable comp-op comparable
+comparable   = number / string-literal /              ; primitive ...
+               true / false / null /                  ; values only
+               path                                   ; path value
```

Comment:
I'm curious, are we planning on supporting literal complex values (i.e. object / arrays) at some point later?  Maybe this needs more discussion.

Reply:
I agree it needs more discussion, but let's defer such discussion for now to keep this PR nice and small.